### ISSUE TITLE
How to Disable/Enable Teleport at runtime with Destination Location

### ISSUE THREAD

**Issue by zogster** [thestonefox/VRTK#1257]
Hi,

this issue is similar to https://github.com/thestonefox/VRTK/issues/1144 but not exactly I guess.

I got a scene with all the markers visible and active, one game object per marker (marker = VRTK_DestinationPoint ).
At runtime, I wanna set some active, some not ... and vice versa.
Every marker has a different destination location (think "you move from one room to another").

Once I re-enable a marker and I trigger the teleport, I get teleported to the location of the marker transform itself and not the destination location.

If I use enableTeleport instead of SetActive, it seems to work, but the marker is always visible.
Is this intended behavior?
I know I can fix it, by hiding the geometry of the marker ... but I somehow think this isn't supposed to happen.

Thanks!

**Comment by thestonefox** [comment]
You need to fill in the Issue template with steps to reproduce

**Comment by zogster** [comment]
That's what you get for not reading ... sorry :)

### Environment

 * Source of VRTK: Github
 * Version of VRTK: My colleague pulled today, enough (he already left)?
 * Version of Unity3D: 5.6.1p1
 * Hardware used: Simulator

### Steps to reproduce
1. Use this unity package [vrtk_test.zip](https://github.com/thestonefox/VRTK/files/1058035/vrtk_test.zip)
2. Open Dev VRTK.scene
3. Look around, see nothing.
4. Press space >> this sets the marker active
5. look at the marker until you teleport

### Expected behavior
1. you Should be on the small cube outside
If you open the script DestinationPointTest.cs and disable the SetActive(false), it works.

### Current behavior
1. Notice that you are ON the marker instead of the destination location

**Comment by thestonefox** [comment]
You need to provide steps to reproduce in an example scene not in one of your own custom scenes.

**Comment by zogster** [comment]
Please understand, I am not our VR programmer.
I am not familiar with your SDK examples.
I don't have time to dig into that. I tried to look at the gaze demo, but no chance (Editor/Simulator).
I used the setup our VR programmer (who dug into your SDK) gave me and I noticed this weird behavior.
I do not have any more time to spend on this issue ... you cannot isolate it more than in the example project I provided.

For all the others:
My current workaround is to let the root game objects always be active.
I disable teleport with enableTeleport and set the locked cursor visible/invisible:

`public static void SetMarkerVisible(VRTK.VRTK_DestinationPoint marker, bool visible)
{
	marker.enableTeleport = visible;
	marker.lockedCursorObject.transform.GetChild(0).gameObject.SetActive( visible );
}`

**Comment by thestonefox** [comment]
Without steps to reproduce in an example scene means whoever looks at this issue has to determine whether the problem is with VRTK or with your custom code. As you can appreciate that people are doing this in their spare time and for free and cannot put time aside to look over custom code.

**Comment by zogster** [comment]
Hi once more,
I have a bit of time further investigating this issue.
You say I have to reproduce the issue with one of your examples.

I need an example where you use the DestinationLocation property of the DestinationPoint prefab.

The closest I came to was "044_CameraRig_RestrictedTeleportZones"
"A scene that uses the VRTK_DestinationPoint prefab to set up a collection of pre-defined teleport locations."

How much modifications are allowed?
I don't wanna waste my time creating another example which won't get looked at.

What I think I need to change:
1) Assign one of the DestinationPoint a DestinationLocation (Let me ask again: Is this feature tested elsewhere?)
2) Create a script, which enables/disables the DestinationPoint.

Would you be good with these changes?

Thanks

**Comment by thestonefox** [comment]
Just use `044_CameraRig_RestrictedTeleportZones` as your base, then provide steps on how to turn that scene into an example that breaks.

Also include your script for enabling/disabling and say where to place it.

Give simple step by step instructions that anyone could follow, even someone without programming experience.

**Comment by zogster** [comment]
### Environment

 * Source of VRTK: Github
 * Version of VRTK: SHA-1: 8155d9447fa8a3050b55895e1b6d80ddb03bd436
 * Version of Unity3D: Unity 5.6.1p1 (64-bit)
 * Hardware used: None, Simulator.
 * SDK used: None, Simulator.

### Steps to reproduce
1. Open Scene "044_CameraRig_RestrictedTeleportZones.unity "
2. Create an empty GameObject and position it somewhere you like. This will be your Destination Location. Name this object "Destination"
3. In the scene, select the Game Object "DestinationPoint".
4. Drag the new created Game Object ("Destination") onto the "Destination Location" property of the Game Object "DestinationPoint"
5. Add the script/component "Destination Point Test" to the Game Object "DestinationPoint" (important! Not "Destination").
    * This script will set the owning game object inactive at start and activate it with one second delay IF you UNCHECK "Set Active On Start".

Testing:
1. Start the scene with "Set Active On Start" CHECKED.
1. Use the DestinationPoint you picked. You will see, that you get teleported to the Destination Location, just as you should.
1. Stop the scene.
1. UNCHECK "Set Active On Start".
1. Use the DestinationPoint you picked. You will see, that you will NOT get teleported to the Destination Location, but to the DestinationPoint directly.


### Expected behavior
It should not affect the DestinationLocation feature if you deactivate a DestinationPoint and then reactivate it again.
If a DestinationLocation is assigned, you should get teleported to that.

### Current behavior
If you deactivate a DestinationPoint, then reactivate it AND you have a DestinationLocation assigned, this doesn't work anymore. You get teleported to the DestinationPoint-Transform, not the DestinationLocation-Transform.



Either download the script file from here:
[DestinationPointTest.zip](https://github.com/thestonefox/VRTK/files/1121377/DestinationPointTest.zip)

Or paste it yourself:
```cs
using UnityEngine;

public class DestinationPointTest : MonoBehaviour
{
    public bool setActiveOnStart = true;
    private GameObject marker;

    void Start()
    {
        marker = this.gameObject;
        marker.SetActive( setActiveOnStart );

        Invoke("SetActiveDelayed", 1);
    }

    void SetActiveDelayed()
    {
        Debug.LogWarning( "DestinationPointTest: Activating game object." );
        marker.SetActive( true );
    }

}
```

**Comment by thestonefox** [comment]
This looks like it's been fixed in the release branch `release/3.3.0-alpha` as the way the destination point works has been changed internally.

I can confirm the issue does exist in `3.2.1` and the github `master` branch but the fix that is in `3.3.0` can't be back ported to the previous version.

Therefore, I'd suggest trying the `3.3.0-alpha` branch and see if it solves your problem.

**Comment by zogster** [comment]
Thanks for the update, I'll test it sometime this week!
